test/conformance/helpers/channel_tracing_test_helper.go

```diff
@@ -23,58 +23,95 @@ import (
 	"time"
 
 	"github.com/openzipkin/zipkin-go/model"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/util/uuid"
 	"knative.dev/eventing/test/base/resources"
 	"knative.dev/eventing/test/common"
 	tracinghelper "knative.dev/eventing/test/conformance/helpers/tracing"
 	"knative.dev/pkg/test/zipkin"
 )
 
-type setupFunc func(t *testing.T, channel string, client *common.Client, loggerPodName string, incomingTraceId bool) (tracinghelper.TestSpanTree, string)
+// SetupClientFunc sets up the client for running tracing tests. It does the equivalent of
+// client.Setup().
+type SetupClientFunc func(*common.Client) error
 
-func tracingTestHelper(t *testing.T, channelTestRunner common.ChannelTestRunner, setup setupFunc) {
-	testCases := map[string]struct {
-		incomingTraceId bool
-		istio           bool
-	}{
+// SetupClientFuncNoop is a SetupClientFunc that does nothing.
+var SetupClientFuncNoop SetupClientFunc = func(*common.Client) error {
+	return nil
+}
+
+// SetupInfrastructureFunc sets up the infrastructure for running tracing tests. It returns the
+// expected trace as well as a string that is expected to be in the logger Pod's logs.
+type SetupInfrastructureFunc func(t *testing.T, channel *metav1.TypeMeta, client *common.Client, loggerPodName string, tc TracingTestCase) (tracinghelper.TestSpanTree, string)
```

Comment:
Ditto.
Also for the following long func defs.

Reply:
Done.